File: lab10/Bank Account Command Pattern/BankApp/src/Query/ShowTransactionHistoryQuery.java
package Query;

import App.bank;

public class ShowTransactionHistoryQuery implements iQuery {
    private bank bank;
    private String accountNumber;

    public ShowTransactionHistoryQuery(bank bank, String accountNumber) {
        this.bank = bank;
        this.accountNumber = accountNumber;
    }

    @Override
    public void query() {
        bank.showTransactionHistory(accountNumber);
    }
}

File: Lab_10/BankApp/Queries/ShowTransactionHistoryQuery.cs
using BankApp.App;

namespace BankApp.Queries;

public class ShowTransactionHistoryQuery : IQuery
{
    private Bank _bank;
    private string _accountNumber;

    public ShowTransactionHistoryQuery(Bank bank, string accountNumber)
    {
        _bank = bank;
        _accountNumber = accountNumber;
    }

    public void Query()
    {
        _bank.ShowTransactionHistory(_accountNumber);
    }
}